NitroxClient/Communication/Packets/Processors/FireDousedProcessor.cs

```diff
@@ -22,11 +25,37 @@ public FireDousedProcessor(IPacketSender packetSender)
         /// </summary>
         public override void Process(FireDoused packet)
         {
-            GameObject fireGameObject = NitroxEntity.RequireObjectFrom(packet.Id);
+            Optional<GameObject> gameObject = NitroxEntity.GetObjectFrom(packet.Id);
+            if (!gameObject.HasValue)
+            {
+                // cannot find that ID. try to douse something at the same location.
+
+                Optional<GameObject> parentObject = NitroxEntity.GetObjectFrom(packet.ParentId);
+                Validate.IsPresent(parentObject, $"Found neither fire {packet.Id} nor parent {packet.ParentId}");
+
+                // look for closest fire
+                Vector3 firePos = new Vector3(packet.xpos, packet.ypos, packet.zpos);
+                float maxDist = float.MaxValue;
+
+                foreach (Fire candidate in parentObject.Value.GetComponentsInChildren<Fire>())
+                {
+                    float candidateDistance = firePos.DistanceSqrXZ(candidate.transform.position);
+                    if (candidateDistance < maxDist)
+                    {
+                        maxDist = candidateDistance;
+                        gameObject = Optional.Of(candidate.gameObject);
+                    }
+                }
+
+                Validate.IsPresent(gameObject, $"Did not find any fires in parent {packet.ParentId}");
+
+                NitroxId replacedId = NitroxEntity.GetId(gameObject.Value);
+                Log.Info($"FireDousedProcessor: Using substitute fire {replacedId} instead of nonexistant {packet.Id}");
+            }
 
             using (packetSender.Suppress<FireDoused>())
             {
-                fireGameObject.RequireComponent<Fire>().Douse(packet.DouseAmount);
+                gameObject.Value.GetComponent<Fire>().Douse(packet.DouseAmount);
```

Comment:
I think we still want RequireComponent so this doesn't null pointer on missing (hard to debug vs Required)

Reply:
Ack.
Hard to navigate when IntelliSense does not show up the out-of-scope extensions...